### ISSUE TITLE
Confusing error message for a struct member used as default parameter value

### ISSUE THREAD

**Issue by AlekseyTs** [dotnet/roslyn#66049]
```
struct S1
{
    int p1;
    
    // error CS1673: Anonymous methods, lambda expressions, query expressions, and local functions inside structs cannot access instance members of 'this'. Consider copying 'this' to a local variable outside the anonymous method, lambda expression, query expression, or local function and using the local instead.
    void M() { void local(int x = p1){} local(); }
}

class C2
{
    int p1;
    
    // error CS1736: Default parameter value for 'x' must be a compile-time constant
    void M() { void local(int x = p1){} local(); }
}
```

Expected: The same error as the one reported for the same situation in a class

**Comment by jcouv** [comment]
Took a quick peek. The problem that one error is reported by `BindParameterDefaultValue` and the other by `ReportDefaultParameterErrors` in the following code (inside `MakeDefaultExpression`).
If `BindParameterDefaultValue` reports a `ERR_ThisStructNotInAnonMeth` then we don't get to `ReportDefaultParameterErrors` to report `ERR_DefaultValueMustBeConstant`.

Since we need to bind the default value to check it's valid (`IsValidDefaultValue`) I'm not sure how we could improve this.

```
            parameterEqualsValue = binderForDefault.BindParameterDefaultValue(defaultSyntax, this, diagnostics, out var valueBeforeConversion);
            if (valueBeforeConversion.HasErrors)
            {
                return ConstantValue.Bad;
            }

            BoundExpression convertedExpression = parameterEqualsValue.Value;
            bool hasErrors = ParameterHelpers.ReportDefaultParameterErrors(binder, ContainingSymbol, parameterSyntax, this, valueBeforeConversion, convertedExpression, diagnostics);
            if (hasErrors)
            {
                return ConstantValue.Bad;
            }
```
